src/layer/domain/router/module/fetch/xhr.ts

```diff
@@ -32,7 +32,7 @@ export function xhr(
   return new AtomicPromise<Either<Error, FetchResponse>>(resolve => {
     if (key && memory.has(key)) return resolve(Right(memory.get(key)!(displayURL, requestURL)));
     const xhr = new XMLHttpRequest();
-    void xhr.open(method, requestURL.path, true);
+    void xhr.open(method, requestURL.reference, true);
```

Comment:
This is insecure. Can you detect add-on context?

Reply:
I figured out this method:

```js
const isInSandbox = typeof global === 'object' && global !== window // true in environments like Firefox add-on context
```